events/patc.py

```diff
@@ -0,0 +1,160 @@
+
+    :return:
+        A list of events
+
+    :rtype:
+        list
+    """
+    res = make_request_content(link)
+
+    # there appears to be faulty calendar events that are unable to be parsed
+    try:
+        results = {
+            "Event Name": res.findAll("th")[-1].getText().strip(),
+            "Description": res.findAll("p")[-1].getText().strip(),
+        }
+    except IndexError:
+        return
+
+    for para in res.findAll("p"):
+        if any([word + ":" in para.getText() for word in KEYWORDS]):
+            logging.info("found event info: %s", para.getText().strip())
+            header = mi.one(para.findAll("strong"))
+            category, _, _ = header.getText().rpartition(":")
+            value = mi.last(para.children)
+            # sometimes BS4 will return NavigableString or some subclass
+            # that appears to be string, we need to encode them to remove
+            # non-ascii characters and then decode them from binary
+            if not isinstance(value, str):
+                value = value.getText()
+            results[category] = value.encode(
+                "ascii", "ignore").decode().strip()
+    results["www"] = URL_BASE + link 
+    return results
+
+
+def format_df_time_columns(df):
+    """Standalone function to format data in accordance to this structure:
+    https://github.com/DataKind-DC/capital-nature-ingest/blob/master/event_schema.md
+
+    :param pandas.DataFrame df:
+        Dataframe to format
+
+    :return:
+        Formatted dataframe
+
+    :rtype:
+        pandas.DataFrame
+    """
+    cols_mapper = {
+        "start_date": "Start Date",
+        "start_time": "Start Time",
+        "end_date": "Start Date",
+    }
+
+    for key, value in cols_mapper.items():
+        if "date" in key:
+            df[key] = df[value].apply(
+                lambda x: parse(x).date().strftime('%Y-%m-%d'))
+        else:
+            df[key] = df[value].apply(
+                lambda x: parse(x).strftime("%H:%M:%S"))
+    return df
+
+
+def main():
+    records = []
+    events = make_request_content(CALENDAR_BASE_URL)
+    for event in events.findAll("a"):
+        # impute empty string because you cannot 
+        # do an equality check on NoneType
+        event_link = event.attrs.get("href", "")
+        if "calendar.aspx" in event_link:
+            url = posixpath.split(
+                event_link)[-1]
+            logging.info("found calendar event, %s", url)
+            data = find_event_data(posixpath.join("Custom", url))
+            # there appears to be faulty 
+            # calendar events that are unable to be parsed
+            if data:
+                records.append(data)
+    data = pd.DataFrame.from_records(records)
+    data = (
+        data.assign(
+            all_day_event=False,
+            time_zone="America/New_York",
+            end_time="23:59:59",
+            event_cost="0",
+            currency="$",
+            venue="Please refer to website",
```

Comment:
The `Event Venue` field needs to contain the event venue and can't say "refer to the website". For this, you could probably scrape the domain name from the `Click Here` href or parse it from the event name itself.